src/abstractdataframe/selection.jl

```diff
@@ -469,8 +479,52 @@ SELECT_ARG_RULES =
     It is not allowed to pass `renamecols=false` if `old_column` is empty
     as it would generate an empty column name.
 
-    Column renaming and transformation operations can be passed wrapped in
-    vectors (this is useful when combined with broadcasting).
+    # Rules when `new_column_name` is a vector of `Symbol` or a string or is `AsTable`
+
+    In this case it is assumed that `fun` returns multiple columns.
+
+    If `fun` returns one of `AbstractDataFrame`, `NamedTuple`, `DataFrameRow`,
+    `AbstractMatrix` then rules described below for `args` being a function or
+    a type apply.
+
+    If `fun` returns an `AbstractVector` then each element of this vector must
```

Comment:
`keys` is called both on the vector itself and on its entries, so it's confusing.

Also, it would be useful to mention here that automatic names are ignored when provided manually. Otherwise one may wonder whether the description applies to this case.

Reply:
>  it would be useful to mention here that automatic names are ignored when provided manually

It was there below. Now I have moved it up, but it means I have to repeat the description twice (but I think it is OK, so I followed your suggestion)

> `keys` is called both on the vector itself and on its entries, so it's confusing.

`keys` is not called on a vector. It is only called on all of its entries.

Essentially what we do is making sure that each entry of the vector has the same schema (to avoid the problem that people repeatedly report in Tables.jl that the schema is taken form the first entry and reused, so wrong results are silently produced if the schema is not constant across all entries of a row table)

Can you please highlight which part of the description is confusing?

Also there is a special case when the vector is empty, in which case we do not call `keys` at all but fall back to `Tables.columntable`, but I would not mention it as it is a corner case that probably is not very important (though it leads to inconsistencies in some corner cases, see https://github.com/JuliaData/Tables.jl/issues/204).

Reply:
Sorry I guess I misread.